### PR TITLE
Implement `mul`

### PR DESCRIPTION
**Pull request by mrmr1993**

### CODE REVIEW

File: optimism/src/mips/interpreter.rs [hunk 1]
```diff
@@ -899,7 +912,19 @@ pub fn interpret_rtype<Env: InterpreterEnv>(env: &mut Env, instr: RTypeInstructi
...
-        RTypeInstruction::MultiplyToRegister => (),
+        RTypeInstruction::MultiplyToRegister => {
+            let rs = env.read_register(&rs);
+            let rt = env.read_register(&rt);
+            let res = {
+                // FIXME: Constrain
+                let pos = env.alloc_scratch();
+                unsafe { env.mul_signed_witness(&rs, &rt, pos) }
+            };
+            env.write_register(&rd, res);
```
Comment (dannywillems) [inline optimism/src/mips/interpreter.rs:923]: Do you have the spec for MUL?
I would have expected to write in HI/LO.

**Review by dannywillems — COMMENTED** [review]
See comment

File: optimism/src/mips/interpreter.rs (hunk 1 above)
Comment (mrmr1993) [reply optimism/src/mips/interpreter.rs:923]: You're thinking of `mult`, it's a different instruction.

Comment (mrmr1993) [reply optimism/src/mips/interpreter.rs:923]: Oh no, *I'm* thinking of the wrong instruction. Lemme fix.

Comment (mrmr1993) [reply optimism/src/mips/interpreter.rs:923]: Actually, wrong again. This `MultiplyToRegister` is indeed correct :man_facepalming:

**Review by dannywillems — APPROVED** [review]

**Pull request merged**